test/extensions/stats_sinks/dog_statsd/config_test.cc

```diff
@@ -31,7 +30,7 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, DogStatsdConfigLoopbackTest,
                          TestUtility::ipTestParamsToString);
 
 TEST_P(DogStatsdConfigLoopbackTest, ValidUdpIp) {
-  const std::string name = StatsSinkNames::get().DogStatsd;
+  const std::string name = "envoy.stat_sinks.dog_statsd";
```

Comment:
Nit: I'm not sure how this fell out of the other PRs but it'd be better to reference same the string constant rather than repeating the string constants in multiple places. OTOH these are mostly in tests. If nothing else you could make this name be member variable of the test-class.

My preference would be to have the string-constant only appear once in the code overall. But I'm not too hung up about it.

Reply:
Yeah I was a bit torn while reviewing the other PRs as well, while I don't like the repeated strings it also felt like requiring a constant in some but not all cases would result in inconsistent code, where a constant would sometimes be used when not necessary if the author cargo culted code that used a constant

Maybe other @envoyproxy/maintainers have thoughts here?

Reply:
The other thing I noted is that the API as it stands needs to copy the string every name `name()` is called. Not sure if that's a concern.

Reply:
The main point of removing well_known_names is that it created a shared dependency between extensions which I think is pretty strange and prevents extensions from being completely self contained and self documented. I agree that string duplication is not awesome, but I think in this case it's preferable to the shared dep. If we see a string being repeated many times, I think it would be OK to have a header file shared only between the extension itself and its tests.

Reply:
Yeah in looking at this PR I would it minimum have the string once in the test, or if you want to share it in a header for just the extension that is fine. Thank you.

/wait

Reply:
Use const value in test header file ?

Reply:
What could I do to make it go ahead? @mattklein123 @jmarantz

Reply:
const value in test header file sgtm.

Reply:
+1 let's at least share the name in the tests.

/wait

Reply:
Done.

How to merge main branch right? When I use `git rebase`, it result in force-push, when I use `git merge`, it cover too many outside commits :(